### PR TITLE
Fix e2e test not load the right image

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

From the e2e-main log, we found that the kustomize not exists, then new built image will not be loaded successfully, this may cause test error like https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/414/pull-kueue-test-e2e-main/1595440995208531968#1:build-log.txt%3A299

```
Image: "gcr.io/k8s-staging-kueue/kueue:v0.3.0-devel-81-g2437b7f" with ID "sha256:03dc0be9da9af117f7ac63c8d968e29c966393c08da99d5a18475e0102dc4762" not yet present on node "kind-control-plane", loading...
./hack/e2e-test.sh: line 21: /home/prow/go/src/sigs.k8s.io/kueue/bin/kustomize: No such file or directory
```

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: hack/e2e-test.sh [hunk 1]
```diff
@@ -1,29 +1,33 @@
 export KUSTOMIZE=$PWD/bin/kustomize
 export GINKGO=$PWD/bin/ginkgo
 export KIND=$PWD/bin/kind
```
Comment (alculquicondor) [inline hack/e2e-test.sh:3]: Let's add `set -e` so that the script fails on any failure.

Comment (kerthcet) [reply hack/e2e-test.sh:23]: updated

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**